### PR TITLE
Minimal smoke test for distributed training

### LINKED ISSUE #32: Minimal smoke test for distributed training
We need a minimal E2E test for distributed training just like we do for the TF operator.

The E2E test should do something like the following
- Create N replicas
- Ensure work can be assigned and performed by all replicas

In the TFJob test case we just do a simple matrix multiple.

P1 because this is a blocker to including the PyTorch operator in our 0.2 release

### PR DESCRIPTION
**Pull request by johnugeorge**
Fixes #32

This includes send and recv from master to worker to test distributed communication

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pytorch-operator/37)

### CODE REVIEW

File: examples/pytorch_sample/build_and_push.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/python
```
Comment (jose5918) [inline examples/pytorch_sample/build_and_push.py]: @johnugeorge Can we use the current build process for this image? Right now we are using the gcloud container builder to build the pytorch-operator container and we can use that for this Dockerfile as well https://github.com/kubeflow/pytorch-operator/blob/master/scripts/build.sh#L40

That way we can test any changes to this smoke test in our CI system

Comment (jose5918) [reply examples/pytorch_sample/build_and_push.py]: Something like what is being done in this new PR https://github.com/kubeflow/pytorch-operator/pull/38/files

**Comment by johnugeorge** [comment]
@jose5918 Updated the script to add the image

File: examples/pytorch_sample/pytorchjob.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+spec:
+  backend: "tcp"
+  masterPort: "23456"
+  replicaSpecs:
+    - replicas: 1
+      replicaType: MASTER
+      template:
+        spec:
+          containers:
+          - image: gcr.io/refined-byte-127219/pytorch-smoke-test:1.0
```
Comment (jose5918) [inline examples/pytorch_sample/pytorchjob.yaml]: You aren't actually using the container you are building.  I think it should be `gcr.io/kubeflow-ci/pytorch-smoke-test:1.0`

File: examples/pytorch_sample/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,4 @@
+FROM pytorch/pytorch:latest
```
Comment (jose5918) [inline examples/pytorch_sample/Dockerfile]: Please us a container version instead of latest

File: examples/pytorch_sample/pytorch_smoke.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+def init_processes(fn, backend='tcp'):
+    """ Initialize the distributed environment. """
+    dist.init_process_group(backend)
+    fn()
+
+def main():
+    logging.info("Torch version: %s", torch.__version__)
+    
+    config_json = os.environ.get("TF_CONFIG", "{}")
```
Comment (jose5918) [inline examples/pytorch_sample/pytorch_smoke.py]: Don't need TF_CONFIG in here. We should probably remove it from the operator anyways

**Pull request merged**